Grinnell-Events-iOS/Grinnell-Events-iOS/ViewController.m

```diff
@@ -75,33 +75,65 @@ - (void)viewDidLoad
             self.filteredEventsDictionary = theEvents;
             // Sort the keys by date
             NSArray *keys = [theEvents allKeys];
+            NSLog(@"%@ is the first key", keys.firstObject);
             self.sortedDateKeys =  [keys sortedArrayUsingComparator: ^(NSString *d1, NSString *d2) {
                 NSDate *date1 = [NSDate dateFromString:d1];
                 NSDate *date2 = [NSDate dateFromString:d2];
                 return [date1 compare:date2];
             }];
+            NSLog(@"%@ is the first sorted key", self.sortedDateKeys.firstObject);
+            NSLog(@"%@ is the last sorted key", self.sortedDateKeys.lastObject);
             
             [self filterContentForSearchText:@"" scope:
              [[self.searchDisplayController.searchBar scopeButtonTitles] objectAtIndex:[self.searchDisplayController.searchBar selectedScopeButtonIndex]]];
             
             [self.tableView reloadData];
-            
+            //set dateformatter in order to convert string to date
+            NSString *dateString = self.sortedDateKeys.firstObject;
+            NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+            dateFormatter.dateFormat = @"MM/dd/yyyy";
+            NSDate *firstDate = [dateFormatter dateFromString:dateString];
             // Set start and end dates in dayPicker
-            NSDate *firstDate = [NSDate dateFromString: self.sortedDateKeys.firstObject ];
+            //commented out this line of code because the date format was not set correctly
+            //NSDate *firstDate = [NSDate dateFromString: self.sortedDateKeys.firstObject ];
+            NSLog(@"First date is %@", firstDate.description);
             NSDate *lastDate = [NSDate dateFromString:self.sortedDateKeys.lastObject];
```

Comment:
So the lastDate is still using the old way to fetch... Is it because only the first date is misformatted? Or is it just because this variable is not used rn?

Reply:
Oops, yeah it's because this variable is not being used right now, will change that.